tests/unit-tests/src/test/java/org/apache/activemq/artemis/tests/unit/core/client/impl/ReadableLargeMessageController.java

```diff
@@ -0,0 +1,877 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ * <br>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <br>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.activemq.artemis.tests.unit.core.client.impl;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.nio.channels.GatheringByteChannel;
+import java.nio.channels.ScatteringByteChannel;
+
+import io.netty.buffer.ByteBuf;
+import org.apache.activemq.artemis.api.core.ActiveMQBuffer;
+import org.apache.activemq.artemis.api.core.SimpleString;
+import org.apache.activemq.artemis.core.client.ActiveMQClientLogger;
+import org.apache.activemq.artemis.core.client.impl.ClientConsumerInternal;
+import org.apache.activemq.artemis.core.client.impl.LargeMessageControllerImpl;
+import org.apache.activemq.artemis.utils.ByteUtil;
+import org.apache.activemq.artemis.utils.DataConstants;
+import org.apache.activemq.artemis.utils.UTF8Util;
+
+final class ReadableLargeMessageController extends LargeMessageControllerImpl implements ActiveMQBuffer {
```

Comment:
This was part of the API Contract. and you moved it to tests only.

Reply:
I see that we expose `LargeMessageController` just on:
```java
public interface ClientLargeMessageInternal extends ClientMessageInternal {

   void setLargeMessageController(LargeMessageController controller);

   LargeMessageController getLargeMessageController();
```
So, if users are not allowed to use internal classes this shouldn't be a big deal (yet to be verified)

Reply:
Not allowed is a strong word.. Users are discouraged to do it..

the original intention of the controller was to be exposed, however this was never achieved.

I think we should have the class you moved to the Testsuite sitting on the same package, in case this breaks someone usage they have the option to decorate the controller with the class you moved, with a big warning they should avoid doing it.